lib/models/device.coffee

```diff
@@ -249,6 +249,41 @@ exports.getApplicationName = (uuid, callback) ->
 	exports.get(uuid).get('application_name').nodeify(callback)
 
 ###*
+# @summary Get application container information
+# @name getApplicationInfo
+# @public
+# @function
+# @memberof resin.models.device
+#
+# @param {String} uuid - device uuid
+# @fulfil {Object} - application info
+# @returns {Promise}
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6').then(function(appInfo) {
+# 	console.log(appInfo);
+# });
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6', function(error, appInfo) {
+# 	if (error) throw error;
+# 	console.log(appInfo);
+# });
+###
+exports.getApplicationInfo = (uuid, callback) ->
+	exports.get(uuid).then (device) ->
```

Comment:
Well, the proxy has the purpose of being the intermediary between the client and the supervisor running on the device. 

The benefit of having it there is that this logic, albeit minimum, will not be duplicated in various projects (e.g: the Python SDK), and gives the proxy the freedom to add/remove/edit warning/deprecation notices at will, without needing to force clients to upgrade their SDKs, or having to commit and publish new versions for little benefit.

We can either hardcode a warning in a single SDK, or, with a bit more work, have a robust and scalable way to report warnings/deprecations to all clients, which would be valuable in the future.